server/queue_priv.php

```diff
@@ -3,12 +3,89 @@
 	require_once "$root/server/config.php";
 	$conn = $DB_CONN;
 
-	switch($_GET['action']) {
-		case 'display': displayAppointment($_GET['id']); break;
-		case 'cancel': cancelAppointment($_GET['id']); break;
+	// TODO: Make it so this is testable
+	require_once "$root/server/user.class.php";
+	$USER = new User();
+	if (!$USER->isLoggedIn()) {
+		header("Location: /unauthorized");
+		die();
+	}
+
+	switch($_REQUEST['action']) {
+		case 'display': displayAppointment($_REQUEST['id']); break;
+		case 'cancel': cancelAppointment($_REQUEST['id']); break;
+		case 'checkIn': checkIn($_REQUEST['time'], $_REQUEST['id']); break;
+		case 'completePaperwork': completePaperwork($_REQUEST['time'], $_REQUEST['id']); break;
+		case 'appointmentStart': appointmentStart($_REQUEST['time'], $_REQUEST['id']); break;
+		case 'appointmentComplete': appointmentComplete($_REQUEST['time'], $_REQUEST['id']); break;
+		case 'appointmentIncomplete': appointmentIncomplete($_REQUEST['explanation'], $_REQUEST['id']); break;
 		default: break;
 	}
 
+	function checkIn($time, $id) {
+		$stmt = $GLOBALS['conn']->prepare(
+			"INSERT INTO ServicedAppointment (timeIn, appointmentId)
+					VALUES (?, ?)"
+		);
+
+		$stmt->execute(array($time, $id));
+		$appointment = $stmt->fetchAll(PDO::FETCH_ASSOC);
+		echo json_encode($appointment);
+		$stmt = null;
+	}
+
+	function completePaperwork($time, $id) {
+		$stmt = $GLOBALS['conn']->prepare(
+			"UPDATE ServicedAppointment
+			SET ServicedAppointment.timeReturnedPapers = ?
+			WHERE ServicedAppointment.appointmentId = ?"
+		);
+
+		$stmt->execute(array($time, $id));
+		$appointment = $stmt->fetchAll(PDO::FETCH_ASSOC);
+		echo json_encode($appointment);
+		$stmt = null;
+	}
+
+	function appointmentStart($time, $id) {
+		$stmt = $GLOBALS['conn']->prepare(
+			"UPDATE ServicedAppointment
+			SET ServicedAppointment.timeAppointmentStarted = ?
+			WHERE ServicedAppointment.appointmentId = ?"
+		);
+
+		$stmt->execute(array($time, $id));
+		$appointment = $stmt->fetchAll(PDO::FETCH_ASSOC);
+		echo json_encode($appointment);
```

Comment:
Similar here about removing those lines, and similarly for all the UPDATEs in this file.